### ISSUE TITLE
area_indexer() is not calculating frequencies properly

### ISSUE THREAD

**Issue by j-harbin** [casaultb/azmpdata#67]
I did a test case for nitrate at station P5, and the result for `plot_availability()` is as follows:

``` r
library(azmpdata)
plot_availability(areaType="station", areaName="P5", parameters="nitrate")
```

![Screen Shot 2021-09-20 at 2 09 18 PM](https://user-images.githubusercontent.com/58750538/134044431-c98a7111-dcf8-4819-a66e-26b839ff3af0.png)

I then did two types of tests, 1) Compare against `area_indexer()` to check if it was a problem with `plot_availability()` or `area_indexer()` and 2) Manually calculate the frequency myself (ie. don't use `area_indexer()`. The results are as follows:

``` r
library(azmpdata)
#> casaultb/azmpdata status:
#>  (Package ver: 0.2019.0.9100) Up to date
#>  (Data ver:2021-01-14 ) Up to date
data("Discrete_Occupations_Stations")
# Step 1 : Check to see if plot_availability() is the same as area_indexer()
 k <- area_indexer(areaTypes="station", areanames="P5", doParameters=T, doMonths=T, parameters="nitrate", fuzzyParameters = F)

## Test December 2001, which as shown by Figure 1, we expect to be 0
d <- k[which(k$year == 2001 & k$month == 12),]
length(d$area)
#> [1] 0

## Test February, 2020, which as shown by Figure 1, we expect to be 1
dd <- k[which(k$year == 2020 & k$month == 02),]
length(dd$area)
#> [1] 1

# Step 2: Check frequency without using area_indexer()
k <- area_indexer(areaTypes="station", areanames="P5", doParameters=T, doMonths=T, parameters="nitrate", fuzzyParameters = F)
unique(k$datafile) # This tells us all data comes from Discrete_Occupations_Stations
#> [1] "Discrete_Occupations_Stations"
data("Discrete_Occupations_Stations")
dos <- Discrete_Occupations_Stations[which(Discrete_Occupations_Stations$year == 2020 & Discrete_Occupations_Stations$month == 2 & Discrete_Occupations_Stations$station == "P5"),]
unique(dos$year) # shows only 2020 being considered
#> [1] 2020
unique(dos$month) # shows only February (2) being considered
#> [1] 2
unique(dos$station) # shows only at P5
#> [1] "P5"
length(dos$nitrate[which(!is.na(dos$nitrate))]) # shows we would expect 5 and not 1
#> [1] 5
```

<sup>Created on 2021-09-20 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

In conclusion, this analysis suggests that the frequency error is not a result of `plot_availability()`, but rather, `area_indexer()`. This makes sense because `plot_availability()` USES `area_indexer()` to perform its functions, so I was not surprised that `plot_availability()` and `area_indexer()` had the same results.

**Comment by j-harbin** [comment]
Hi All,

I'm just using this as a place to track my debugging. The following code gives me the same answers, which suggest the problem is with `doParameters` and not subsetting by year and month.

``` r
library(azmpdata)
#> casaultb/azmpdata status:
#>  (Package ver: 0.2019.0.9100) Up to date
#>  (Data ver:2021-01-14 ) Up to date
k <- area_indexer(areaType="station", areanames = "P5", parameters="nitrate",
                  doParameters = T, fuzzyParameters = F, doMonths = T)

message(length(!is.na(k$year[which(k$year == 2020 & k$month == 02)])))
#> 1

kk <- area_indexer(areaType="station", areanames = "P5", parameters="nitrate",
                   doParameters = T, fuzzyParameters = F, doMonths = T,
                   months=02,year=2020)

message(length(kk$year))
#> 1
```

<sup>Created on 2021-09-27 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

**Comment by j-harbin** [comment]
Another example we are investigating is the following (@MMMcMahon I'll direct this comment to you instead of copying all of the code into teams)

``` r
library(azmpdata)
#> casaultb/azmpdata status:
#>  (Package ver: 0.2019.0.9100) Up to date
#>  (Data ver:2021-01-14 ) Up to date
plot_availability(areaType="station", areaName="HL2", parameters="chlorophyll_0_100")
```

![](https://i.imgur.com/SzN0ADW.png)

<sup>Created on 2021-10-13 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

Specifically, I'll look at May (05) in 2001. `plot_availability()` suggests a frequency of 5, and manually suggests 5.

``` r
library(azmpdata)
#> casaultb/azmpdata status:
#>  (Package ver: 0.2019.0.9100) Up to date
#>  (Data ver:2021-01-14 ) Up to date
k <- area_indexer(areanames="HL2", months = 05, years=2001, parameters = "hlorophyll_0_100")
k # area_indexer() also shows 5
#>   year area section station         parameter month cnt
#> 1 2001   NA      NA     HL2 chlorophyll_0_100     5   5
#>                       datafile
#> 1 Derived_Occupations_Stations
unique(k$datafile) # shows to investigate Derived_Occupations_Stations
#> [1] "Derived_Occupations_Stations"
data(Derived_Occupations_Stations)
d <- Derived_Occupations_Stations$chlorophyll_0_100[which(Derived_Occupations_Stations$station == "HL2" & Derived_Occupations_Stations$year == 2001 & Derived_Occupations_Stations$month == 05)]
length(which(!is.na(d)))
#> [1] 5
```

<sup>Created on 2021-10-13 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>
To me is seems all is good here?

**Comment by j-harbin** [comment]
Closing this issue and both Mike and I agree that `plot_availability()` and `area_indexer()` are calculating the correct frequencies.
